website/content/en/preview/getting-started/getting-started-with-karpenter/cloudformation.yaml

```diff
@@ -35,48 +35,224 @@ Resources:
-              - ec2:DescribeSubnets
-              - pricing:GetProducts
-              - ssm:GetParameter
-          - Effect: Allow
-            Action:
-              # Write Operations
-              - sqs:DeleteMessage
-              # Read Operations
-              - sqs:GetQueueAttributes
-              - sqs:GetQueueUrl
-              - sqs:ReceiveMessage
-            Resource: !GetAtt KarpenterInterruptionQueue.Arn
-          - Effect: Allow
-            Action:
-              - iam:PassRole
-            Resource: !Sub "arn:${AWS::Partition}:iam::${AWS::AccountId}:role/KarpenterNodeRole-${ClusterName}"
-          - Effect: Allow
-            Action:
-              - eks:DescribeCluster
-            Resource: !Sub "arn:${AWS::Partition}:eks:${AWS::Region}:${AWS::AccountId}:cluster/${ClusterName}"
+      # The PolicyDocument must be in JSON string format because we use a StringEquals condition that uses a interpolated
+      # value in one of its key parameters which isn't natively supported by CloudFormation
+      PolicyDocument: !Sub |
+        {
+          "Version": "2012-10-17",
+          "Statement": [
+            {
+              "Sid": "AllowScopedEC2InstanceActions",
+              "Effect": "Allow",
+              "Resource": [
+                "arn:${AWS::Partition}:ec2:${AWS::Region}::image/*",
+                "arn:${AWS::Partition}:ec2:${AWS::Region}::snapshot/*",
+                "arn:${AWS::Partition}:ec2:${AWS::Region}:${AWS::AccountId}:spot-instances-request/*",
+                "arn:${AWS::Partition}:ec2:${AWS::Region}:${AWS::AccountId}:security-group/*",
+                "arn:${AWS::Partition}:ec2:${AWS::Region}:${AWS::AccountId}:subnet/*",
+                "arn:${AWS::Partition}:ec2:${AWS::Region}:${AWS::AccountId}:launch-template/*"
+              ],
+              "Action": [
+                "ec2:RunInstances",
+                "ec2:CreateFleet"
+              ]
+            },
+            {
+              "Sid": "AllowScopedEC2LaunchTemplateActions",
+              "Effect": "Allow",
+              "Resource": "arn:${AWS::Partition}:ec2:${AWS::Region}:${AWS::AccountId}:launch-template/*",
+              "Action": "ec2:CreateLaunchTemplate",
+              "Condition": {
+                "StringEquals": {
+                  "aws:RequestTag/kubernetes.io/cluster/${ClusterName}": "owned"
+                },
+                "StringLike": {
+                  "aws:RequestTag/karpenter.sh/provisioner-name": "*"
+                },
+                "ForAllValues:StringEquals": {
+                  "aws:TagKeys": [
+                    "Name",
+                    "karpenter.sh/managed-by",
+                    "karpenter.sh/provisioner-name",
+                    "kubernetes.io/cluster/${ClusterName}",
+                    "karpenter.k8s.aws/cluster"
+                  ]
+                }
+              }
+            },
+            {
+              "Sid": "AllowScopedEC2InstanceActionsWithTags",
```

Comment:
With the above `AllowScopedEC2InstanceActions` Statement, isn't this one unnecessary?

Reply:
There is a _slight_ difference. Maybe we can discuss if it makes sense to consolidate them. If you were to combine these two statements, then you would require the request that is generated by the RunInstances call to the LaunchTemplate to _require_ the `kubernetes.io/cluster/<cluster-name>` request tag. In my testing, I wasn't able to get this to work since whatever request is generated to the launch template by fleet doesn't seem to propagate the tag, even if I specify the tag in the tagSpecifications (maybe @bwagner5 knows more about this). 

By separating it out, I'm able to allow the CreateFleet/RunInstances call to succeed to make its request to the launch template while also enforcing that whichever launch templates that Karpenter creates must have the relevant tags

Reply:
> you would require the request that is generated by the RunInstances call to the LaunchTemplate to require the kubernetes.io/cluster/<cluster-name> request tag.

I do think this is something we want right?

> I wasn't able to get this to work since whatever request is generated to the launch template by fleet doesn't seem to propagate the tag, even if I specify the tag in the tagSpecifications 

We may need to follow up with EC2 folks on this.

Reply:
Yeah, I think so. If it's not propagating right now, do you think it's fine to iterate on this template moving forward?